src/os/bluestore/BlueStore.cc

```diff
@@ -3218,13 +3218,24 @@ void BlueStore::DeferredBatch::_audit(CephContext *cct)
 BlueStore::Collection::Collection(BlueStore *ns, Cache *c, coll_t cid)
   : CollectionImpl(cid),
     store(ns),
-    osr(new OpSequencer(store)),
     cache(c),
     lock("BlueStore::Collection::lock", true, false),
     exists(true),
     onode_map(c)
 {
-  osr->shard = cid.hash_to_shard(ns->m_finisher_num);
```

Comment:
Cosmetic: `ns` is quite surprising name for a `BlueStore` instance, I guess. Maybe rename the ctor's input parameter to `store`, `_store` or something similar?